modules/standard/BigInteger.chpl

```diff
@@ -2414,39 +2414,39 @@ When ``numer/denom`` does not produce an integer, this method may produce incorr
   /*
     .. warning::
 
-       d is deprecated - please use div
+       bigint.divisible_p is deprecated, use bigint.isDivisible instead
   */
   pragma "last resort"
   deprecated
-  "d is deprecated - please use div"
-  proc bigint.divisible_p(const ref d: bigint) : int {
-    return this.divisible_p(div=d);
+  "bigint.divisible_p is deprecated, use bigint.isDivisible instead"
+  proc bigint.divisible_p(const ref d: bigint) : bool {
```

Comment:
In the deprecated method, we are only altering the return type, if I let it be `int` as it was, then it would automatically convert our `bool` values to `int(64)` as that is what it is supposed to return as per the function definition

Reply:
Yeah, I think that's what we want.  Ideally, we want the presence of a new deprecation message to be the only thing to have changed for the user when a function they use first becomes deprecated.  This allows them time to adjust their code to use the replacement (if one exists) without breaking their code in the meanwhile.  This isn't always possible, but it's something we should strive for.  While the return type of this function might convert nicely into an int in code written like this:
```chapel
var x: int = foo.divisible_p(bar);
```
it will change the type of code written like this:
```chapel
var x = foo.divisible_p(bar);
```
and they might be relying on x's type being int elsewhere in the code.  We want to give them time to adjust their code intentionally, and not change program behavior without telling them in ways that could be difficult to track down

Reply:
Oh Yes, that makes sense! I'll keep the original return type only, then for the deprecated method.
Bu... how do we let the people know that the return type is no longer `int`, should I add something in the deprecation message?

Reply:
You can and it would probably be friendly to do so, but I think it's also fair to rely on users noticing it when looking at the new function, since in this case the return type is explicitly declared as part of the method declaration.  If the program were inferring the return type and it had changed, I would feel more strongly about noting it (since it would be more subtle and difficult to notice)

Reply:
Ohk, so I will let it be and not add any extra statements for now.
Cool... so now we just need to finalise the name of those 2 methods!